app/helpers/automated_tests_helper.rb

```diff
@@ -380,6 +474,435 @@ def parse_test_output(repo_assignment_dir, assignment, logfile, data)
     # Return parsed (or unparsed) test output
     output
   end
+
+
+
+
+
+
+  def AutomatedTestsHelper.request_a_test_run(grouping_id, call_on, current_user)
+    @current_user = current_user
+    #@submission = Submission.find(submission_id)
+    @grouping = Grouping.find(grouping_id)
+    @assignment = @grouping.assignment
+    @group = @grouping.group
+
+    @repo_dir = File.join(MarkusConfigurator.markus_config_automated_tests_repository, @group.repo_name)
+    export_group_repo(@group, @repo_dir)
+
+    @list_run_scripts = scripts_to_run(@assignment, call_on)
+
+    async_test_request(grouping_id, call_on)
+  end
+
+
+  # Export group repository for testing. Students' submitted files
+  # are stored in the group svn repository. They must be exported
+  # before copying to the test server.
+  def self.export_group_repo(group, repo_dir)
+    # Create the automated test repository
+    if !(File.exists?(MarkusConfigurator.markus_config_automated_tests_repository))
+      FileUtils.mkdir(MarkusConfigurator.markus_config_automated_tests_repository)
+    end
+
+    # Delete student's assignment repository if it already exists
+    delete_repo(repo_dir)
+
+    # export
+    return group.repo.export(repo_dir)
+    rescue Exception => e
+      return "#{e.message}"
+  end
+
+  # Delete repository directory
+  def self.delete_repo(repo_dir)
+    # Delete student's assignment repository if it already exists
+    if File.exists?(repo_dir)
+      FileUtils.rm_rf(repo_dir)
+    end
+  end
+
+
+  # Find the list of test scripts to run the test. Return the list of
+  # test scripts in the order specified by seq_num (running order)
+  def self.scripts_to_run(assignment, call_on)
+    # Find all the test scripts of the current assignment
+    all_scripts = TestScript.where(assignment_id: assignment.id)
+
+    list_run_scripts = Array.new
+
+    # If the test run is requested at collection (by Admin or TA),
+    # All of the test scripts should be run.
+    if call_on == "collection"
+      list_run_scripts = all_scripts
+    else
+      # If the test run is requested at submission or upon request,
+      # verify the script is allowed to run.
+      all_scripts.each do |script|
+        if (call_on == "submission") && script.run_on_submission
+          list_run_scripts.insert(list_run_scripts.length, script)
+        elsif (call_on == "request") && script.run_on_request
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.